ComputeClient/Compute.Client.UnitTests/Snapshot/SnapshotAccessorTests.cs

```diff
@@ -0,0 +1,115 @@
+			Assert.IsFalse(essentialsSnapshotServicePlan.supportsReplication);
+			Assert.IsFalse(essentialsSnapshotServicePlan.available);
+
+			Assert.AreEqual("ONE_MONTH", oneMonthSnapshotServicePlan.id);
+			Assert.AreEqual("One Month: 7d-4w", oneMonthSnapshotServicePlan.displayName);
+			Assert.AreEqual("Daily Snapshots retained for 7 Days, Weekly Snapshots retained for 31 Days", oneMonthSnapshotServicePlan.description);
+			Assert.AreEqual("DAILY", oneMonthSnapshotServicePlan.snapshotFrequency);
+			Assert.IsFalse(oneMonthSnapshotServicePlan.supportsReplication);
+			Assert.IsTrue(oneMonthSnapshotServicePlan.available);
+
+			Assert.AreEqual("THREE_MONTH", threeMonthSnapshotServicePlan.id);
+			Assert.AreEqual("Three Month: 14d-12w", threeMonthSnapshotServicePlan.displayName);
+			Assert.AreEqual("Daily Snapshots retained for 14 Days, Weekly Snapshots retained for 90 Days", threeMonthSnapshotServicePlan.description);
+			Assert.AreEqual("DAILY", threeMonthSnapshotServicePlan.snapshotFrequency);
+			Assert.IsFalse(threeMonthSnapshotServicePlan.supportsReplication);
+			Assert.IsTrue(threeMonthSnapshotServicePlan.available);
+
+			Assert.AreEqual("TWELVE_MONTH", twelveMonthSnapshotServicePlan.id);
+			Assert.AreEqual("Twelve Month: 31d-12w-12m", twelveMonthSnapshotServicePlan.displayName);
+			Assert.AreEqual("Daily Snapshots retained for 31 Days, Weekly Snapshots retained for 90 Days, Monthly Snapshots retained for 1 Year", twelveMonthSnapshotServicePlan.description);
+			Assert.AreEqual("DAILY", twelveMonthSnapshotServicePlan.snapshotFrequency);
+			Assert.IsFalse(twelveMonthSnapshotServicePlan.supportsReplication);
+			Assert.IsTrue(twelveMonthSnapshotServicePlan.available);
+		}
+
+		[TestMethod]
+		public async Task ListSnapshotServicePlans_WithFiltersAndPagingOptions()
+		{
+			var filter = new SnapshotServicePlanListOptions();
+			filter.Available = true;
+			filter.Id = Guid.NewGuid();
+
+			var pagingRequest = new PageableRequest();
+			pagingRequest.PageNumber = 1;
+			pagingRequest.PageSize = 5;
+
+			var expectedRelativeUriPath = string.Format(ApiUris.ListSnapshotServicePlans(accountId) + "?available={0}&id={1}&pageSize={2}&pageNumber={3}", "true", filter.Id, pagingRequest.PageSize, pagingRequest.PageNumber);
+			var expectedUri = new Uri(expectedRelativeUriPath, UriKind.Relative);
+
+			requestsAndResponses.Add(expectedUri, RequestFileResponseType.AsGoodResponse("ListSnapshotServicePlansWithFilterResponse.xml"));
+
+			var client = GetWebApiClient();
+			var accessor = new SnapshotAccessor(client);
+
+			var snapshotServicePlansPaginated = await accessor.ListSnapshotServicePlans(filter, pagingRequest);
+
+			Assert.IsNotNull(snapshotServicePlansPaginated);
+			Assert.IsNotNull(snapshotServicePlansPaginated);
+			Assert.AreEqual(1, snapshotServicePlansPaginated.pageCount);
+			Assert.AreEqual(1, snapshotServicePlansPaginated.totalCount);
+			Assert.AreEqual(10, snapshotServicePlansPaginated.pageSize);
```

Comment:
Should the pageSize be 5 as specified in the pagingRequest.PageSize?

Reply:
Good spot will clean up. The test marshalls back the file content of  _ListSnapshotServicePlansWithFilterResponse.xml_ thats where the pageSize in the actual response is defined.